api/server/handlers/argo/resource_hooks.go

```diff
@@ -0,0 +1,45 @@
+package argo
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/porter-dev/porter-agent/api/server/config"
+	"github.com/porter-dev/porter-agent/api/server/types"
+	"github.com/porter-dev/porter-agent/pkg/controllers"
+	"github.com/porter-dev/porter/api/server/shared"
+)
+
+// ResourceHookHandler contains helper functions for listening to Argo CD Resource Hook events
+type ResourceHookHandler struct {
+	decoderValidator shared.RequestDecoderValidator
+	resultWriter     shared.ResultWriter
+
+	Config                 *config.Config
+	ArgoCDResourceConsumer controllers.ArgoCDResourceHookConsumer
+}
+
+// NewResourceHookHandler returns a new instance of ResourceHookHandler
+func NewResourceHookHandler(config *config.Config, consumer controllers.ArgoCDResourceHookConsumer) ResourceHookHandler {
+	return ResourceHookHandler{
+		resultWriter:           shared.NewDefaultResultWriter(config.Logger, config.Alerter),
+		decoderValidator:       shared.NewDefaultRequestDecoderValidator(config.Logger, config.Alerter),
+		Config:                 config,
+		ArgoCDResourceConsumer: consumer,
+	}
+}
+
+// ServeHTTP implements Go's HTTP handler interface for listening to ArgoCD resource events
+func (h ResourceHookHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var req types.ArgoCDResourceHook
+
+	if ok := h.decoderValidator.DecodeAndValidate(w, r, &req); !ok {
+		return
+	}
+
+	err := h.ArgoCDResourceConsumer.Consume(r.Context(), req)
+	if err != nil {
```

Comment:
We use the `apierrors` package from porter to ensure consistent logging/formatting of internal errors -- see [here](https://github.com/porter-dev/porter-agent/blob/60b0d6994d92c49fa344dfe128ab88f0b518a014/api/server/handlers/incident/get.go#L44) for reference.